### ISSUE TITLE
[FEA] Implement Series isin

### ISSUE THREAD

**Issue by beckernick** [rapidsai/cudf#1577]
**Is your feature request related to a problem? Please describe.**
We should implement the `isin` method for Series (and eventually DataFrames). This [method](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Series.isin.html) returns "a boolean Series showing whether each element in the Series matches an element in the passed sequence of values exactly." This is very useful for filtering with Boolean indexing.

I'd like to be able to do something like the following:

```python
import pandas as pd
b = pd.DataFrame({'y':range(10), 'x':range(10,20)})
c = pd.DataFrame({'y':range(5, 15), 'x':range(8,18)})

b[b.y.isin(c.x)]
	y	x
8	8	18
9	9	19
```

This feature request would include support for `isin` on both Numerical and StringColumns.

**Comment by venatir** [comment]
+1

**Comment by kkraus14** [comment]
This is resolved as of 0.9.
